src/securityPolicy.c

```diff
@@ -365,13 +368,12 @@ security_policy_t policyForSignTxWitness(
 	if (isSigningPoolRegistrationAsOwner) {
 		DENY_UNLESS(is_valid_stake_pool_owner_path(pathSpec));
 	} else {
-		// TODO Perhaps we can relax this?
-		WARN_UNLESS(has_reasonable_account_and_address(pathSpec));
-
-		// TODO deny this? or check for depth in is_valid_witness?
-		WARN_IF(is_too_deep(pathSpec));
+		// witness for input or withdrawal
```

Comment:
also stake delegation/key deregistration is included in this branch which do require respective staking key witnesses, I would just drop the comment as it's rather self-evident.

I would also consider removing the else branch and apply the `DENY_IF(is_too_deep(pathSpec))` regardless of isSigningPoolRegistrationAsOwner value. The check is already done in the is_valid_stake_pool_owner_path so it's technically redundant but also the final code is simpler imho. Sorry, I should have realized in the previous refactoring suggestion I had